### PR TITLE
WIP : Allow Multiple Selected Note to delete

### LINKED ISSUE #15885: Multiselect Note type and delete
> Selecting multiple note type (just like in pc) to delete
https://forums.ankiweb.net/t/delete-unused-note-types/42394/

💡 Click and hold note type will display a trash icon in top right to delete

### PR DESCRIPTION
**Pull request by neeldoshii**
## Fix
* Fixes #15885

## Todo

- [ ] 1. Deletion seems to work properly, but if a new note is added it doesn't considers old notes it only works on the selection works on newly added note only. Somehow the getSelectedItems() doesn't return old cards selected value.
- [ ] 2. Enhance the ui (remove checkbox & change background color as discussed in discord).
- [ ] 3. Add snackbar to show "x selected cards are deleted" for feedback to user.
- [ ] 4. Refactor unused code for review.

### Working of Multiple selection and allowing to delete

https://github.com/ankidroid/Anki-Android/assets/60827173/a230fafe-a6c0-4c8a-ae40-3249b84792f2

### Current bug in the code

https://github.com/ankidroid/Anki-Android/assets/60827173/05757290-1dea-4064-a0d7-e8145690053b

 

## Checklist
_Please, go through these checks before submitting the PR._

- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/res/values/02-strings.xml [hunk 1]
```diff
@@ -433,4 +433,5 @@ opening the system text to speech settings fails">
     <string name="card_browser_unavailable_when_notes_mode">Unavailable in ‘Notes’ mode</string>
 
     <string name="card_template_reposition_template" comment="move a card template to a new position">Reposition</string>
+    <string name="title_delete" comment="content description (title) of the delete icon in menu">Delete</string>
```
Comment (SanjaySargam) [inline AnkiDroid/src/main/res/values/02-strings.xml:436]: `Delete` string already exists, so remove this string

File: AnkiDroid/src/main/java/com/ichi2/anki/notetype/ManageNotetypes.kt [hunk 2]
```diff
@@ -125,9 +127,25 @@ class ManageNotetypes : AnkiActivity() {
                 return true
             }
         })
+        // Menu for deleting multiple notes
```
Comment (neeldoshii) [inline AnkiDroid/src/main/java/com/ichi2/anki/notetype/ManageNotetypes.kt:130]: Reviewer Note:

Is it okay having do menu inflate or lets get it into one?

**Comment by neeldoshii** [comment]
Remove Tag : Conflict, Strings

> Deletion seems to work properly, but if a new note is added it doesn't considers old notes it only works on the selection works on newly added note only. Somehow the getSelectedItems() doesn't return old cards selected value.

@david-allison can you help me where I am going wrong with logic? I am unable to fix this.

**Comment by david-allison** [comment]
Could you rephrase? I don't understand the problem

**Comment by neeldoshii** [comment]
I have a video in description : working and when bug occurs.

1. The normal multiple deletion works properly.
2. The logic for deletion gets failed when you add a new note type. for ex in the bugged video added note type : test1, test2. But the. 3 selected were not deleted only recently added new note was deleted.

**Comment by david-allison** [comment]
@neeldoshii 

Time to improve your debugging skills and see if you can replicate this screenshot:

Screenshot 2024-04-05 at 20 12 14

The screenshot also shows the issue: `notetypesAdapter.currentList` only has one `isSelected`, whereas the UI shows 3.

**Comment by neeldoshii** [comment]
Don't have much time atm busy recently. Tag : [Needs a new dev] @david-allison
